main_comment.go

```diff
@@ -147,30 +147,40 @@ func processGitHubComment(
 //	}
 func parsePrOptions(commentBody string) (map[string]string, error) {
 	prRepos := make(map[string]string)
-	var err error = nil
-
-	words := strings.Fields(commentBody)
-	tokensCount := len(words)
-	for i, w := range words {
-		if w == "--pr" && i < (tokensCount-1) {
-			v := words[i+1]
-			o := strings.Index(v, "/")
-			if o > 0 {
-				skip := 0
-				if v[o:][0] == '/' && len(v[o:]) > 1 {
-					skip = 1
-				}
-				revision := v[o+skip:]
-				if _, err := strconv.Atoi(revision); err == nil {
-					revision = "pull/" + revision + "/head"
+	var err error
+
+	prs := strings.Split(commentBody, "--pr")[1:] // without "start pipeline "
+	for _, userInput := range prs {
+		userInput := strings.TrimSpace(userInput)
+		if len(userInput) == 0 {
+			err = errors.New("Parse error: Empty user input near --pr")
+			return nil, err
+		}
+		userInputParts := strings.Split(userInput, "/")
+
+		if len(userInput) > 0 {
+			var revision string
+			switch len(userInputParts) {
+			case 2: // we can have both deviceauth/1 and mender/3.1.x syntax
+				// repo/<pr_number> syntax
+				if _, err := strconv.Atoi(userInputParts[1]); err == nil {
+					revision = "pull/" + userInputParts[1] + "/head"
+				} else {
+					// feature branch
+					prPartsAndRandString := strings.Split(userInputParts[1], " ")
+					revision = prPartsAndRandString[0]
```

Comment:
I think iterating over the words as in the original version is better, mainly because the new version makes it hard to add new arguments. What if we add an `--enterprise` argument? It also leads to splitting on spaces in the above line, long after you already split on `--pr` further above, which is a bit unexpected. It seems more natural to split on word boundaries first, and then inspect each word separately.

Otherwise I agree with the semantics. This is the behavior we want.

Reply:
I came back to iterating over words.